chips/sam4l/src/pm.rs

```diff
@@ -259,243 +148,170 @@ pub enum SystemClockSource {
     /// Use the internal digital frequency locked loop (DFLL) sourced from
     /// the internal RC32K clock. Note this typically requires calibration
     /// of the RC32K to have a consistent clock.
-    DfllRc32k,
+    DfllRc32kAt48MHz,
 
     /// Use an external crystal oscillator as the direct source for the
     /// system clock.
-    ExternalOscillator,
+    ExternalOscillatorAt16MHz,
 
     /// Use an external crystal oscillator as the input to the internal
     /// PLL for the system clock. This expects a 16 MHz crystal.
-    ExternalOscillatorPll,
+    PllExternalOscillatorAt48MHz,
+}
+
+/// Which frequency range does your external oscillator fall in? Configuration
```

Comment:
On the SAM4L, the gain for external oscillators needs to be set, which is a different value for ranges of oscillators. (see manual Page 259, Table 13-6)  For example, Gain 2 works for oscillators in the 4-8 MHz range.

I will definitely add this information in the comments.